src/pcre2_match.c

```diff
@@ -5792,6 +5792,8 @@ fprintf(stderr, "++ %2ld op=%3d %s\n", Fecode - mb->start_code, *Fecode,
 
       /* Disable compiler warning. */
       offset = 0;
```

Comment:
"Disable compiler warning" is a bit ironic :)

Clang now complains about the fact that we're writing a dead value to a variable, when there are no further reads from it!

Reply:
Since multiple of such cases are updated: is there a better way to do this? Separating smart and stupid compilers in some way?

Reply:
Doing `(void)variable` in 5 or ten places is the just the tax you pay to support multiple compilers, and avoid generating warnings on all of them. Ugly, but harmless.

For what it's worth: Clang is being clever, not stupid. Assigning to a variable, but never reading the value back, is a very helpful warning that can catch typos when you have variables with similar names.